### ISSUE TITLE
Add note to docs that private fields of structs cannot be fuzzed using this package

### ISSUE THREAD

**Issue by ericcornelissen** [google/gofuzz#44]
I tried to use this package (through go-fuzz) for the first time when I was fuzzing a struct that has only private fields. IIt took me a while to figure out that this package doesn't put any values into private fields...

Now, I'm not terribly familiar with reflection in Go but from what I can tell based on the reflection docs and some testing is that it would be impossible for this package to set the value of private fields. Am i correct?

If the above statement is correct, I think it would be useful if there was a note in the README (or somewhere else in the docs) stating that private fields cannot be populated by this package. Thoughts?

**Comment by lavalamp** [comment]
Yes, you're correct, and I agree a note in the README seems useful.

**Comment by ericcornelissen** [comment]
I'm not sure where in the README you would want to add this, but I think having a note somewhere stating something as simple as:

_"gofuzz cannot set the value of private fields."_

Would suffice.

Maybe after the first example with a struct like so:

---

...
Customize the chance of getting a nil pointer:

```
f := fuzz.New().NilChance(.5)
var fancyStruct struct {
  A, B, C, D *string
}
f.Fuzz(&fancyStruct) // About half the pointers should be set.
```

> :information_source: **Note**: gofuzz cannot set the value of private fields.

You can even customize the randomization completely if needed:
...

---

Or just at the end of the document, so that it's easy in the future to add more notes like this one:

---

...
Happy testing!

## Caveats

- gofuzz cannot set the value of private fields.
---
